ui/prometheus-plugin/src/plugins/prometheus-time-series-query/PrometheusTimeSeriesQueryEditor.tsx

```diff
@@ -24,6 +25,9 @@ export function PrometheusTimeSeriesQueryEditor(props: PrometheusTimeSeriesQuery
   const { onChange, value } = props;
   const { datasource } = value;
 
+  // TODO: Need to get the Prometheus URL here from the selected datasource
+  const promURL = 'https://demo.promlabs.com';
```

Comment:
@LukeTillman is there a simple way to get the selected datasource config? Ideally, an API like

```tsx
const datasourceName = 'my-ds'
const datasourceSpec = useDatasourceSpec<PromDatasourceSpec>(datasourceName)
const url = datasourceSpec.instanceURL
```

Reply:
This is the current API for getting datasources:

https://github.com/perses/perses/blob/40c96a7ae6a6f7d97df9bc0457278e82a209b573/ui/plugin-system/src/runtime/datasources.ts#L18-L31

You can see by that `TODO` that I was thinking about getting rid of the method for getting the spec completely in favor of just getting a "client" object (which is currently just "whatever client object a datasource wants to use" and doesn't have any kind of shared interface at all).

I think what I was picturing was trying to adapt our `PrometheusClient` object to work with Code Mirror? IIRC, the PromQL codemirror packages will let you pass a custom client object (that matches some specific interface) that then gets used. Pretty sure we actually ended up using that functionality internally at Chronosphere because we wanted to avoid some behavior that the default client in those packages does (e.g. falling back to the Prom series metadata endpoint for label metadata queries, etc.) If you want to follow-up with me, I can probably point you to some of that internally.

I was picturing the creation of that Datasource client object as potentially becoming an injection point for custom behavior like additional HTTP headers, etc. for library usage scenarios like ours. But not something I've totally fleshed out yet.

Reply:
Yeah I think both make sense. I think its still nice to get the spec but also have the ability to consume an instantiated client of a datasource based on the spec configuration. Even if it was something like:

```tsx
const datasourceClient = useDatasourceClient('my-ds');

// extends DatasourceClient which contains certain properties
datasourceClient.spec // returns the JSON spec